src/test/userRepository.test.ts

```diff
@@ -0,0 +1,166 @@
+import bcrypt from "bcrypt";
+import mongoose from "mongoose";
+import { ZodError } from "zod";
+import { FIELD_NAMES, ERROR_MESSAGES } from "../constants";
+import { UserModel } from "../models";
+import { userRepository } from "../repositories";
+import { closeConnectionDatabase, connectDatabase } from "../utils";
+
+const MOCK_EMAIL = "mail@mail.com";
+const MOCK_PASSWORD = "password";
+
+function validateDate(date: NativeDate): boolean {
+  const convertedDate = new Date(date);
+  return !isNaN(convertedDate.getTime());
+}
+
+describe("create user", () => {
+  beforeAll(async () => {
+    await connectDatabase();
+  });
+
+  beforeEach(async () => {
+    await UserModel.deleteMany({});
+  });
+
+  afterAll(async () => {
+    await UserModel.deleteMany({});
+    await closeConnectionDatabase();
+  });
+
+  test("should fail: undefined email", async () => {
+    const [, error] = await userRepository.create({ password: MOCK_PASSWORD });
+
+    // Assert
+    expect(error).toBeInstanceOf(ZodError);
+    const zodError = error as ZodError;
+
+    expect(zodError.issues[0]?.message).toBe(ERROR_MESSAGES.EMAIL_UNDEFINED);
+    expect(zodError.issues[0]?.path).toEqual([FIELD_NAMES.EMAIL]);
+  });
+
+  test("should fail: empty email", async () => {
+    const [, error] = await userRepository.create({
+      email: "",
+      password: MOCK_PASSWORD,
+    });
+
+    // Assert
+    expect(error).toBeInstanceOf(ZodError);
+    const zodError = error as ZodError;
+
+    expect(zodError.issues[0]?.message).toBe(ERROR_MESSAGES.EMAIL_EMPTY);
+    expect(zodError.issues[0]?.path).toEqual([FIELD_NAMES.EMAIL]);
+  });
+
+  test("should fail: invalid email", async () => {
+    const [, error] = await userRepository.create({
+      email: "abc",
+      password: MOCK_PASSWORD,
+    });
+
+    // Assert
+    expect(error).toBeInstanceOf(ZodError);
+    const zodError = error as ZodError;
+
+    expect(zodError.issues[0]?.message).toBe(ERROR_MESSAGES.EMAIL_INVALID);
+    expect(zodError.issues[0]?.path).toEqual([FIELD_NAMES.EMAIL]);
+  });
+
+  test("should fail: undefined password", async () => {
+    const [, error] = await userRepository.create({ email: MOCK_EMAIL });
+
+    // Assert
+    expect(error).toBeInstanceOf(ZodError);
+    const zodError = error as ZodError;
+
+    expect(zodError.issues[0]?.message).toBe(ERROR_MESSAGES.PASSWORD_UNDEFINED);
+    expect(zodError.issues[0]?.path).toEqual([FIELD_NAMES.PASSWORD]);
+  });
+
+  test("should fail: password too short", async () => {
+    const [, error] = await userRepository.create({
+      email: MOCK_EMAIL,
+      password: "321",
+    });
+
+    // Assert
+    expect(error).toBeInstanceOf(ZodError);
```

Comment:
As discussed in the repository file, Zod validation does not belong at the repository level.